jobs/wf2-2-upsertTEIs.js

```diff
@@ -12,69 +12,77 @@ fn(state => {
   };
 });
 
-each('patients[*]', state => {
-  const patient = state.data;
+// Prepare DHIS2 data model for patients
+fn(state => {
   const { genderOptions } = state;
 
-  const [OldId] = patient.identifiers.filter(
-    i => i.identifierType.uuid === '8d79403a-c2cc-11de-8d13-0010c6dffd0f'
-  ); //finding DHIS2 patient_number
+  const pluckIdentifier = (patient, uuid) =>
+    patient.identifiers.find(i => i.identifierType.uuid === uuid);
 
-  const [openMRSID] = patient.identifiers.filter(
-    i => i.identifierType.uuid === '05a29f94-c0ed-11e2-94be-8c13b969e334'
-  ); //finding OpenMRS ID that was auto-assigned
-  const identifier = OldId ? OldId.identifier : openMRSID.identifier;
-  
-  const dateCreated = patient.auditInfo.dateCreated.substring(0,10);
+  const patientsMapping = state.patients.map(patient => {
+    const oldId = pluckIdentifier(
+      patient,
+      '8d79403a-c2cc-11de-8d13-0010c6dffd0f'
+    ); //finding DHIS2 patient_number
+    const openMRSID = pluckIdentifier(
+      patient,
+      '05a29f94-c0ed-11e2-94be-8c13b969e334'
+    ); //finding OpenMRS ID that was auto-assigned
 
-  //   console.log(`Identifier: ${identifier}`);
-  //   console.log(`UUID: ${patient.uuid}`);
-  //   console.log(`Age: ${patient.person.age}`);
-  //   console.log(`Gender: ${genderOptions[patient.person.gender]}`);
+    const identifier = oldId ? oldId.identifier : openMRSID.identifier;
```

Comment:
This is looking better. But looking again this mornig I can't help but think this is a lot simpler and easier to read if you do:
```
const DHIS2_PATIENT_NUMBER='8d79403a-c2cc-11de-8d13-0010c6dffd0f'
const OPENMRS_AUTO_ID='05a29f94-c0ed-11e2-94be-8c13b969e334'

const identifier = patient.identifiers.find(i => i.identifierType.uuid === DHIS2_PATIENT_NUMBER) ||  
   patient.identifiers.find(i => i.identifierType.uuid === OPENMRS_AUTO_ID)
```
The `find` syntax is so simple and clean I really don't think it's worth aliasing, and the uses of constants makes the code a bit easier to read without the need for comments,